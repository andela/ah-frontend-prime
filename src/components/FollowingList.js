import React from "react";

const FollowingList = props => {
  const { following } = props;
  return (
    <div className="modal-dialog" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <h5 className="modal-title">My Following</h5>
          <button
            type="button"
            className="close"
            data-dismiss="modal"
            aria-label="Close"
          >
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
          {following && following.length > 0
            ? following.map(item => (
                <div
                  className="row"
                  id="following"
                  key={item.username}
                  onClick={() => {
                    document.location.href = `/profile/${item.username}`;
                  }}
                  key={item.username}
                >
                  <button className="col col-lg-3" style={{ border: "none" }}>
                    <img
                      src={
                        item.image
                          ? `${item.image}`
                          : "https://cdn.pixabay.com/photo/2013/07/13/10/07/man-156584__340.png"
                      }
                      alt="profile"
                      className="profile-page"
                    />
                  </button>
                  <div className="col col-lg-9">
                    <h5>{item.username}</h5>
                    <p>{item.full_name}</p>
                  </div>
                  <br />
                </div>
              ))
            : "No one is on the following List"}
        </div>
        <div class="modal-footer" />
      </div>
    </div>
  );
};

export default FollowingList;
