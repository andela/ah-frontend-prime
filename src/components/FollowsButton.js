import React from "react";

const FollowsButton = props => {
  const { isfollowing, unfollowUser, followUser } = props;
  return (
    <button
      onClick={
        isfollowing
          ? unfollowUser.bind(this, sessionStorage.getItem("userview_name"))
          : followUser.bind(this, sessionStorage.getItem("userview_name"))
      }
      className="btn btn-outline-success"
    >
      {isfollowing ? "Unfollow" : "Follow"}
    </button>
  );
};

export default FollowsButton;
