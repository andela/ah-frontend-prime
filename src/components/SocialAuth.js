import React from "react";
import FacebookLogin from "react-facebook-login";
import GoogleLogin from "react-google-login";
import config from "../../src/config.json";

export const SocialAuth = props => (
  <div>
    <div className="face-book">
      <FacebookLogin
        appId={config.appId}
        fields="name,email,picture"
        callback={props.facebooksuccess}
        icon={
          <img
            src="https://img.icons8.com/color/96/000000/facebook.png"
            width="=40px"
            height="40px"
            alt="facebook"
          />
        }
        textButton="SIGN IN WITH FACEBOOK"
      />
    </div>
    <br />
    <div className="google">
      <GoogleLogin
        clientId={config.clientId}
        buttonText="SIGN IN WITH GOOGLE"
        onSuccess={props.googlesuccess}
        onFailure={props.googlefailure}
      />
    </div>
  </div>
);

export default SocialAuth;
