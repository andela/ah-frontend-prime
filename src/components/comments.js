import React, { Component } from "react";
import moment from "moment";
import { articleComments, fetchComments } from "../actions/commentsActions";
import { connect } from "react-redux";
import "../styles/comments.scss";


export class Comments extends Component {
  constructor(props) {
    super(props);
    this.state = {
      comments: ""
    };
  }

  componentDidMount() {
    const { fetchComments, articleSlug } = this.props;
    fetchComments(articleSlug);
  }

  handleOnChange = event => {
    event.preventDefault();
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  handleOnSubmit = event => {
    event.preventDefault();
    const { articleSlug, articleComments, fetchComments } = this.props;
    const { comments } = this.state;
    const data = {
      body: comments
    };
    articleComments(data, articleSlug);
    fetchComments(articleSlug);
  };

  render() {
    const { commentsMessage } = this.props;
    const comment = commentsMessage.map(message => (
      <section className="comments" key={message.id}>
        <article className="comment">
          <a className="comment-img" href="#non">
            <img src={message.author.image} alt="" width="50" height="50" />
          </a>
          <div className="comment-body">
            <div className="text">
              <p>{message.body}</p>
            </div>
            <p className="attribution">
              by <a href={`/profile/${message.author.username}`}>{message.author.username}</a> at {moment(message.createdAt).format(
                          "h:mm:ss a [on] MMMM Do YYYY.")}
                        
            </p>
          </div>
        </article>
      </section>
    ));
    return (
      <div className="row">
        <div className="col col-lg-1" />
        <div className="col col-lg-10">
          <div className="row">
            <div className="col col-lg-1" />
            <div className="col col-lg-11">
              <form
                method="POST"
                onSubmit={this.handleOnSubmit}
                className="comment-forms"
              >
                <textarea
                  className="form-control"
                  name="comments"
                  value={this.state.comments}
                  onChange={this.handleOnChange}
                >
                  comments here
                </textarea>
                <button
                  type="submit"
                  className="btn btn-comment"
                  style={{ marginTop: "10px" }}
                >
                  comment
                </button>
              </form>
            </div>
          </div>
          {comment}
        </div>
        <div className="col col-lg-1" />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  commentsSuccess: state.comments.success,
  commentsErrors: state.comments.errors,
  commentsMessage: state.getComments.comments
});
export default connect(
  mapStateToProps,
  { articleComments, fetchComments }
)(Comments);
