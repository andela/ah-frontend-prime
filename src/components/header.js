import React from "react";
import PropTypes from "prop-types";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import InputBase from "@material-ui/core/InputBase";
import { fade } from "@material-ui/core/styles/colorManipulator";
import { withStyles } from "@material-ui/core/styles";
import SearchIcon from "@material-ui/icons/Search";

const styles = theme => ({
  root: {
    width: "100%"
  },
  grow: {
    flexGrow: 1
  },
  menu: {
    marginLeft: -12,
    marginRight: 20
  },
  title: {
    display: "none",
    [theme.breakpoints.up("sm")]: {
      display: "block"
    }
  },
  inputRoot: {
    color: "inherit",
    width: "100%"
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 9,
    transition: theme.transitions.create("width"),
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      width: 118,
      "&:focus": {
        width: 180
      }
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 8,
    height: "100%",
    position: "absolute",
    pointerEvents: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center"
  },
  bar: {
    backgroundColor: "#003153",
    boxShadow: "none"
  },
  search: {
    position: "relative",
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.18),
    "&:hover": {
      backgroundColor: fade(theme.palette.common.white, 0.29)
    },
    marginLeft: 0,
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      marginLeft: theme.spacing.unit,
      width: "auto"
    }
  }
});

function TopAppBar(props) {
  const { classes } = props;
  return (
    <div className={classes.root}>
      <AppBar position="static" className={classes.bar}>
        <Toolbar>
          <Typography
            className={classes.title}
            variant="h6"
            color="inherit"
            noWrap
          >
            Authors Haven
          </Typography>
          <div className={classes.grow} />

          <div className={classes.search}>
            <div className={classes.searchIcon}>
              <SearchIcon />
            </div>
            <InputBase
              placeholder="Searchâ€¦"
              classes={{
                root: classes.inputRoot,
                input: classes.inputInput
              }}
            />
          </div>
          <div className="nav-div">
            <a href="login" className="nav-links">
              Sign In
            </a>
            &nbsp;&nbsp; | &nbsp;&nbsp;
            <a href="register" className="nav-links">
              Get Started
            </a>
          </div>
        </Toolbar>
      </AppBar>
    </div>
  );
}

TopAppBar.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(TopAppBar);
