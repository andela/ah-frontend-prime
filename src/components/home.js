import React, { Component } from "react";
import { connect } from "react-redux";
import "../styles/app.scss";
import moment from "moment";
import { Link } from "react-router-dom";
import { getArticlesAction } from "../actions/getArticles";

export class Home extends Component {
  constructor(props) {
    super(props);
    this.fetchNext = this.fetchNext.bind(this);
    this.fetchPrevious = this.fetchPrevious.bind(this);
  }
  componentDidMount() {
    const { getArticlesAction } = this.props;
    getArticlesAction();
  }
  fetchNext() {
    const { next, getArticlesAction } = this.props;
    getArticlesAction(next);
  }
  fetchPrevious() {
    const { previous, getArticlesAction } = this.props;
    getArticlesAction(previous);
  }
  render() {
    const { articles } = this.props;
    const articleList = articles.length ? (
      articles.map(article => {
        return (
          <div className="article" key={article.slug}>
            <div className="inner">
              <Link
                className="links"
                to={{
                  pathname: "/article/" + article.slug
                }}
              >
                <div className="article-card">
                  <h3>{article.title.substring(0, 30)} ...</h3>
                  <p>
                    <b>{article.description.substring(0, 60)} ...</b>
                  </p>
                  <span>
                    <b>Author: {article.author.username}</b>
                  </span>
                  <br />

                  <span className="card-date">
                    Created at{" "}
                    {moment(article.createdAt).format(
                      "h:mm:ss a [on] MMMM Do YYYY."
                    )}
                  </span>
                  <br />
                  <p>
                    <b>{article.reading_time}</b>
                  </p>
                </div>

                <div>
                  <img
                    src={
                      article.image
                        ? article.image
                        : "https://previews.123rf.com/images/twindesigner/twindesigner1708/twin" +
                          "designer170800135/84202763-ah-brush-letter-logo-design-with-black-" +
                          "circle-creative-brushed-letters-icon-logo-.jpg"
                    }
                    className="img-tile"
                  />
                </div>
              </Link>
            </div>
          </div>
        );
      })
    ) : (
      <div>No Articles in Authors Haven Yet</div>
    );

    return (
      <div className="big-container">
        <div className="articles-top">
          <div className="best-article">
            <div className="inner">
              <img
                src={
                  "http://allpicts.in/download/22916/2018/03/Natural_Images_HD_1" +
                  "080p_Download_with_Waterfall_in_Tollymore_Forest_Park-1440x900.jpg/"
                }
                className="img-top"
              />
              <div className="inner">
                <div className="inner">
                  <div className="best-article-card">
                    <h3>
                      Facebook's Plan to Fuse its Messaging Apps Is Not about
                      Your Privacy Just follow the Money
                    </h3>
                    <span>
                      <b>Author: Yasha Levin in OneZero</b>
                    </span>
                    <br />
                    <span className="card-date">Created a few hours ago</span>
                  </div>
                </div>
              </div>
            </div>
            <div />
          </div>
          <div className="top-articles">{articleList}</div>
          <div className="top-articles">{articleList}</div>
        </div>

        <div className="articles">
          <div className="heading">
            <h3>Featured Articles</h3>
          </div>
          <hr />
          {articleList}
        </div>
        <div className="articles1">
          <div className="heading">
            <h3>Popular Articles</h3>
          </div>
          <hr />
          {articleList}
        </div>
        <div>
          <button
            className="btn btn-outline-success btn eian"
            onClick={() => this.fetchPrevious()}
          >
            Previous
          </button>{" "}
          <button
            className="btn btn-outline-success btn aken"
            onClick={() => this.fetchNext()}
          >
            Next
          </button>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  articles: state.getArticlesReducer.articles,
  next: state.getArticlesReducer.next,
  previous: state.getArticlesReducer.previous
});

export default connect(
  mapStateToProps,
  { getArticlesAction }
)(Home);
