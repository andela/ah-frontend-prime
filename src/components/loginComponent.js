import React from 'react';
import '../styles/app.scss';

const Login = () => {
  return (
    <div>
      <h2>Login</h2>
    </div>
	)
}

export default Login;
