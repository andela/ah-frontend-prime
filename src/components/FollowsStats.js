import React from "react";
import FollowingList from "./FollowingList";
import FollowersList from "./FollowersList";

const FollowsStats = props => {
  const { following_count, followers_count, followers, following } = props;
  return (
    <div className="row">
      <div
        className="modal fade"
        id="exampleModal"
        tabIndex={-1}
        role="dialog"
        aria-labelledby="exampleModalLabel"
        aria-hidden="true"
      >
        <FollowingList following={following} />
      </div>
      <div
        className="modal fade"
        id="exampleModal1"
        tabIndex={-1}
        role="dialog"
        aria-labelledby="exampleModalLabel"
        aria-hidden="true"
      >
        <FollowersList followers={followers} />
      </div>
      <a
        className="col-md-4"
        data-toggle="modal"
        data-target="#exampleModal1"
        href="#"
      >
        {followers_count === 1 ? "Follower" : "Followers"}
        {" : "}
        <button className="stats">
          <strong>{followers_count ? followers_count : 0} </strong>
        </button>
      </a>
      <a
        className="col-md-4"
        data-toggle="modal"
        data-target="#exampleModal"
        href="#"
      >
        Following{" : "}
        <button className="stats">
          <strong>{following_count ? following_count : "0"} </strong>
        </button>
      </a>
    </div>
  );
};

export default FollowsStats;
