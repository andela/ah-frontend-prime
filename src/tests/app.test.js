import React from "react";
import renderer from "react-test-renderer";
import App from "../containers/app";
import LoginComponent from "../components/login/LoginModal";
import { Home } from "../components/home";
import { shallow } from "enzyme";
import data from "./mock_data/moxios_mock";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import store from "../store";
import { BrowserRouter, Route } from "react-router-dom";
import { SocialAuth } from "../components/SocialAuth";

const props = {
  articles: data.articles,
  getArticlesAction: jest.fn()
};

describe("login Component", () => {
  const script = document.createElement("script");
  script.type = "text/javascript";
  document.getElementsByTagName("head")[0].appendChild(script);

  const div = document.createElement("div");
});

describe("App", () => {
  it("should render without crashing", () => {
    const wrapper = renderer.create(<App />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("login Component", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Route>
              <LoginComponent />
            </Route>
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render the home page", () => {
    const wrapper = shallow(<Home {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
