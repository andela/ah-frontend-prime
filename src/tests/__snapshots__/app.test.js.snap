// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App login Component 1`] = `
<div>
  <div
    className="MuiGrid-container-2 MuiGrid-spacing-xs-24-26"
  >
    <div
      className="MuiGrid-item-3 MuiGrid-grid-md-2-60 MuiGrid-grid-lg-2-74"
    />
    <div
      className="MuiGrid-item-3 MuiGrid-grid-md-8-66 MuiGrid-grid-lg-8-80"
    >
      <div
        className="MuiPaper-root-99 MuiPaper-elevation2-103 MuiPaper-rounded-100 Paper"
      >
        <h3>
          Welcome back!
        </h3>
        <p>
          SignIn to like, dislike, favorite and unfavorite various articles,
          <br />
          follow authors and comment on articles.
        </p>
        <div
          className="MuiGrid-container-2 MuiGrid-spacing-xs-24-26"
        >
          <div
            className="MuiGrid-item-3 MuiGrid-grid-md-6-64 MuiGrid-grid-lg-6-78"
          >
            <div
              className="button-collective"
            >
              <div>
                <div>
                  <div
                    className="face-book"
                  >
                    <span
                      style={
                        Object {
                          "opacity": 0.6,
                          "transition": "opacity 0.5s",
                        }
                      }
                    >
                      <button
                        className="kep-login-facebook metro"
                        onClick={[Function]}
                        type="button"
                      >
                        <img
                          alt="facebook"
                          height="40px"
                          src="https://img.icons8.com/color/96/000000/facebook.png"
                          width="=40px"
                        />
                        SIGN IN WITH FACEBOOK
                      </button>
                      <style
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": Array [
                              Array [
                                9,
                                ".kep-login-facebook{font-family:Helvetica,sans-serif;font-weight:700;-webkit-font-smoothing:antialiased;color:#fff;cursor:pointer;display:inline-block;font-size:calc(.27548vw + 12.71074px);text-decoration:none;text-transform:uppercase;transition:background-color .3s,border-color .3s;background-color:#4c69ba;border:calc(.06887vw + .67769px) solid #4c69ba;padding:calc(.34435vw + 13.38843px) calc(.34435vw + 18.38843px)}.kep-login-facebook.small{padding:calc(.34435vw + 3.38843px) calc(.34435vw + 8.38843px)}.kep-login-facebook.medium{padding:calc(.34435vw + 8.38843px) calc(.34435vw + 13.38843px)}.kep-login-facebook.metro{border-radius:0}.kep-login-facebook .fa{margin-right:calc(.34435vw + 3.38843px)}",
                                "",
                              ],
                            ],
                          }
                        }
                      />
                    </span>
                  </div>
                  <br />
                  <div
                    className="google"
                  >
                    <button
                      disabled={true}
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#fff",
                          "border": "1px solid transparent",
                          "borderRadius": 2,
                          "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
                          "color": "rgba(0, 0, 0, .54)",
                          "display": "inline-flex",
                          "fontFamily": "Roboto, sans-serif",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "opacity": 0.6,
                          "padding": 0,
                        }
                      }
                      type="button"
                    >
                      <div
                        style={
                          Object {
                            "background": "#fff",
                            "borderRadius": 2,
                            "marginRight": 10,
                            "padding": 10,
                          }
                        }
                      >
                        <svg
                          height="18"
                          width="18"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="#000"
                            fillRule="evenodd"
                          >
                            <path
                              d="M9 3.48c1.69 0 2.83.73 3.48 1.34l2.54-2.48C13.46.89 11.43 0 9 0 5.48 0 2.44 2.02.96 4.96l2.91 2.26C4.6 5.05 6.62 3.48 9 3.48z"
                              fill="#EA4335"
                            />
                            <path
                              d="M17.64 9.2c0-.74-.06-1.28-.19-1.84H9v3.34h4.96c-.1.83-.64 2.08-1.84 2.92l2.84 2.2c1.7-1.57 2.68-3.88 2.68-6.62z"
                              fill="#4285F4"
                            />
                            <path
                              d="M3.88 10.78A5.54 5.54 0 0 1 3.58 9c0-.62.11-1.22.29-1.78L.96 4.96A9.008 9.008 0 0 0 0 9c0 1.45.35 2.82.96 4.04l2.92-2.26z"
                              fill="#FBBC05"
                            />
                            <path
                              d="M9 18c2.43 0 4.47-.8 5.96-2.18l-2.84-2.2c-.76.53-1.78.9-3.12.9-2.38 0-4.4-1.57-5.12-3.74L.97 13.04C2.45 15.98 5.48 18 9 18z"
                              fill="#34A853"
                            />
                            <path
                              d="M0 0h18v18H0z"
                              fill="none"
                            />
                          </g>
                        </svg>
                      </div>
                      <span
                        style={
                          Object {
                            "fontWeight": 500,
                            "paddingBottom": 10,
                            "paddingLeft": 0,
                            "paddingRight": 10,
                            "paddingTop": 10,
                          }
                        }
                      >
                        SIGN IN WITH GOOGLE
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-item-3 MuiGrid-grid-md-6-64 MuiGrid-grid-lg-6-78 grid-register"
            id="login-form"
          >
            <form
              method="post"
            >
              <div
                className="MuiFormControl-root-126 textField"
              >
                <label
                  className="MuiFormLabel-root-141 MuiFormLabel-required-146 MuiInputLabel-required-134 MuiInputLabel-root-130 MuiInputLabel-formControl-135 MuiInputLabel-animated-138 LoginComponent-floatingLabelFocusStyle-1"
                  data-shrink={false}
                  htmlFor="email"
                >
                  email
                  <span
                    className="MuiFormLabel-asterisk-147"
                  >
                     *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root-161 MuiInput-root-148 MuiInput-underline-152 MuiInputBase-formControl-162 MuiInput-formControl-149"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input-171 MuiInput-input-156"
                    disabled={false}
                    id="email"
                    name="email"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="text"
                  />
                </div>
              </div>
              <div
                className="MuiFormControl-root-126 textField"
                maxLength={16}
                minLength={6}
              >
                <label
                  className="MuiFormLabel-root-141 MuiFormLabel-required-146 MuiInputLabel-required-134 MuiInputLabel-root-130 MuiInputLabel-formControl-135 MuiInputLabel-animated-138 LoginComponent-floatingLabelFocusStyle-1"
                  data-shrink={false}
                  htmlFor="password"
                >
                  password
                  <span
                    className="MuiFormLabel-asterisk-147"
                  >
                     *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root-161 MuiInput-root-148 MuiInput-underline-152 MuiInputBase-formControl-162 MuiInput-formControl-149"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input-171 MuiInput-input-156 MuiInputBase-inputType-174 MuiInput-inputType-159"
                    disabled={false}
                    id="password"
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="password"
                  />
                </div>
              </div>
              <div
                className="loadedContent"
              />
              <button
                className="MuiButtonBase-root-204 MuiButton-root-178 MuiButton-text-180 MuiButton-flat-183 button-success"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="submit"
              >
                <span
                  className="MuiButton-label-179"
                >
                  Login
                </span>
                <span
                  className="MuiTouchRipple-root-207"
                />
              </button>
            </form>
          </div>
        </div>
        <p>
          Don't have an account yet?
           
          <a
            href="/register"
            id="link"
            onClick={[Function]}
          >
            create one now
          </a>
        </p>
        <p>
          Forgot your password?
           
          <a
            href="/passwordresetemail"
            id="link"
            onClick={[Function]}
          >
            reset now
          </a>
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-item-3 MuiGrid-grid-md-2-60 MuiGrid-grid-lg-2-74"
    />
  </div>
</div>
`;

exports[`App should render without crashing 1`] = `
<div>
  <div
    className="Toastify"
  />
  <div>
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <nav
        className="navbar navbar-expand-sm "
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="#"
            >
              <h4>
                <i
                  className="fas fa-bars"
                  id="nav-icon"
                  onClick={[Function]}
                  style={
                    Object {
                      "marginRight": "20px",
                    }
                  }
                />
                <a
                  className="nav-links"
                  href="/"
                >
                  Author's Haven
                </a>
              </h4>
            </a>
          </li>
        </ul>
        <ul
          className="nav navbar-nav ml-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/login"
              onClick={[Function]}
            >
              Sign In
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link get-started "
              href="/register"
              onClick={[Function]}
            >
              Get Started
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="big-container"
    >
      <div
        className="articles-top"
      >
        <div
          className="best-article"
        >
          <div
            className="inner"
          >
            <img
              className="img-top"
              src="http://allpicts.in/download/22916/2018/03/Natural_Images_HD_1080p_Download_with_Waterfall_in_Tollymore_Forest_Park-1440x900.jpg/"
            />
            <div
              className="inner"
            >
              <div
                className="inner"
              >
                <div
                  className="best-article-card"
                >
                  <h3>
                    Facebook's Plan to Fuse its Messaging Apps Is Not about Your Privacy Just follow the Money
                  </h3>
                  <span>
                    <b>
                      Author: Yasha Levin in OneZero
                    </b>
                  </span>
                  <br />
                  <span
                    className="card-date"
                  >
                    Created a few hours ago
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div />
        </div>
        <div
          className="top-articles"
        >
          <div>
            No Articles in Authors Haven Yet
          </div>
        </div>
        <div
          className="top-articles"
        >
          <div>
            No Articles in Authors Haven Yet
          </div>
        </div>
      </div>
      <div
        className="articles"
      >
        <div
          className="heading"
        >
          <h3>
            Featured Articles
          </h3>
        </div>
        <hr />
        <div>
          No Articles in Authors Haven Yet
        </div>
      </div>
      <div
        className="articles1"
      >
        <div
          className="heading"
        >
          <h3>
            Popular Articles
          </h3>
        </div>
        <hr />
        <div>
          No Articles in Authors Haven Yet
        </div>
      </div>
    </div>
  </div>
</div>
`;
